mypy/checker.py

```diff
@@ -7519,6 +7542,29 @@ def has_valid_attribute(self, typ: Type, name: str) -> bool:
     def get_expression_type(self, node: Expression, type_context: Type | None = None) -> Type:
         return self.expr_checker.accept(node, type_context=type_context)
 
+    def check_deprecated(self, typ: SymbolNode | None, context: Context) -> None:
+        """Warn if deprecated and not directly imported with a `from` statement."""
+        if isinstance(typ, Decorator):
+            typ = typ.func
+        if isinstance(typ, (FuncDef, OverloadedFuncDef, TypeInfo)) and (
+            typ.deprecated is not None
+        ):
+            for imp in self.tree.imports:
+                if isinstance(imp, ImportFrom) and any(typ.name == n[0] for n in imp.names):
+                    break
+            else:
+                self.warn_deprecated(typ, context)
+
+    def warn_deprecated(self, typ: SymbolNode | None, context: Context) -> None:
```

Comment:
The parameter name `typ` is now a bit misleading now here as well (I will not add this comment, if there are more similar cases).

Reply:
Also changed. I think there are no other similar cases - I added only two `..._deprecated` methods.